### PR TITLE
Add api design.md

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: content/concepts/managedcluster.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+---
+
+## ManagedCluster Registration
+
+`ManagedCluster` is an cluster scoped API on hub cluster to define a kubernetes cluster registered to the hub cluster. In open-cluster-management, the cluster registration follows a `double opt-in` mechanism which is:
+
+1. An agent can register a cluster to the hub by creating a `ManagedCluster` resource on the hub.
+2. An admin user on the hub needs to explicitly accept the registration of the `ManagedCluster`.
+
+This process ensures that any agent with a bootstrap kubeconfig to hub can request to join as a `ManagedCluster`, while the request must be approved by an admin on the hub.
```
Comment (xuezhaojun) [inline content/concepts/managedcluster.md:13]: `bootstrap` kubeconfig may looks not very explicit to fresh new users

File: content/concepts/addon.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+title: Addons
+weight: 4
+---
+
+## What is an addon?
+
+Open-cluster-managemeent has a mechanism to help developers to develop an extension based on the foundation components for the purpose of working with multiple clusters in various asepcts. The examples of addons includes:
```
Comment (skeeey) [inline content/concepts/addon.md:8]: management

File: content/concepts/managedcluster.md (hunk 1 above)
Comment (qiujian16) [reply content/concepts/managedcluster.md]: updated

File: content/concepts/addon.md (hunk 2 above)
Comment (qiujian16) [reply content/concepts/addon.md]: updated

File: content/concepts/placement.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+title: Placement
+weight: 3
```
Comment (mikeshng) [inline content/concepts/placement.md:3]: `geekdocHidden: true`

maybe add the above if this is just a placeholder file

Comment (mikeshng) [reply content/concepts/placement.md:3]: as discussed with Qiu JIan. content will be populated soon.

**Pull request merged**